cloudinit/config/cc_mounts.py

```diff
@@ -534,7 +534,7 @@ def handle(_name, cfg, cloud, log, _args):
         need_mount_all = True
 
     if need_mount_all:
-        activate_cmds.append(["mount", "-a"])
+        activate_cmds.append(["mount", "-a", "-O", "no_netdev"])
```

Comment:
Interesting option.  The open question is what ensures these nfs mounts get mounted before user-config runs which expects mounts to be up?

From the hip idea; add a drop-in config to cloud-config.service.conf.d which greps /etc/fstab for netdev and if found run mount -a -O _netdev to mount any of those that aren't mounted. 

This might not help any module which run after cc_mount but before cloud-config.service runs... namely:

- set_hostname
- update_hostname
- update_etc_hots
- resolv_conf
- ca-certs
- rsyslog
- users-groups
- ssh

I users/groups and ssh are the most likely hits (if someone is nfs mounting something over /home.